lib/senders/index.js

```diff
@@ -89,12 +123,11 @@ module.exports = function (config, log) {
             uaBrowserVersion: opts.uaBrowserVersion,
             uaOS: opts.uaOS,
             uaOSVersion: opts.uaOSVersion
-          }
-        ))
-      }
-      mailer.sendPasswordChangedNotification = function (email, opts) {
-        return P.resolve(mailer.passwordChangedEmail(
-          {
+          }))
+      },
+      sendPasswordChangedNotification(email, opts) {
+        return bounceGatedMailer(email)
+          .then(mailer => mailer.passwordChangedEmail({
```

Comment:
That is indeed a bad edge-case!

I could add a `catch` to those routes, so that at the least, the response is a 200, and the UI reacts accordingly. That would mean "silently" not notifying the user of a security event... But if it's been hard bouncing, they wouldn't receive it anyways!